libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -1164,15 +968,10 @@ void AC_AutoTune::save_tuning_gains()
     }
 
     // sanity check the rate P values
-    if ((axes_completed & AUTOTUNE_AXIS_BITMASK_ROLL) && roll_enabled() && !is_zero(tune_roll_rp)) {
+    if ((axes_completed & AUTOTUNE_AXIS_BITMASK_ROLL) && roll_enabled() && (!is_zero(tune_roll_rp) || allow_zero_rate_p())) {
```

Comment:
I think here too in "save_tuning_gains()" it is too confusing to have three implementations.  Let's reduce it by two by removing the base class implementation.  In particular I worry that we could easily end up with situation where half the gains are saved because of a small error/mismatch in the sanity checks.

and it would be difficult to determine why.

Reply:
Done